packages/core/src/browser/keybinding.ts

```diff
@@ -11,6 +11,10 @@ import { KeyCode, KeySequence } from './keys';
 import { ContributionProvider } from '../common/contribution-provider';
 import { ILogger } from "../common/logger";
 import { StatusBarAlignment, StatusBar } from './status-bar/status-bar';
+import { CustomError } from '../common/error';
+import { isOSX } from '../common/os';
+
+class KeybindingNotAvailableError extends CustomError { }
```

Comment:
I think it would be better to do this PR without any `CustomError` classes. That should go into a separate PR.
Why?:
 - I do not think, it should be called as `CustomError`.
 - We have not yet decided whether we go with error codes or not.
 - Why would we need that fine granularity of errors, do you handle them and act differently?

Reply:
In tests yes, I do compare errors.

I made a second [PR](https://github.com/theia-ide/theia/pull/1536) that regrouped the application errors, I think you also saw the [issue](https://github.com/theia-ide/theia/issues/1535).

Because it was better to be able to not just catch `Error` in the tests I followed @epatpol advice of doing the 2 PRs and rebasing this one on the one with errors.

Of course I was waiting for input on the implementation. I have uncommitted changes on this PR also relative to the use case around cross-scope collisions and key sequence priority. Apparently there was a little problem prior to my refactoring.

Reply:
@kittaakos I thought GH would handle better a dependency on another patch like Gerrit (where it would clearly show that a patch depends on another, and what changes are part of this one and not the other) but it seems the only way to make this clear is either submit two PRs with no dependencies on each other, or submit one PR with all the changes. The issue with the first approach is that you can't profit from changes made in one in the other one, and the issue with the second approach is that it's sometimes difficult to make a PR very specific without adding too much stuff that may seem unnecessary around it.

All in all the easiest solution is indeed to remove the Error commit and just focus on keymaps for this one.